### PR TITLE
Rename environment variables to not include "DATABRICKS"

### PR DESCRIPTION
**Pull request by prithvikannan**
>[!NOTE]
> The reason we want to rename all of our env vars to not include the substring `DATABRICKS` is because JSL previously was using that as a sign that they were in a Databricks Runtime, which means they wouldn't start their own spark runtime. As a result, whenever env vars with `DATABRICKS` in their name were set in a Databricks model serving environment, JSL models were unusable.
> This has already been fixed in JSL 5.3.6 ([release link](https://github.com/JohnSnowLabs/johnsnowlabs/releases/tag/536)). In order to not break existing JSL customers on Databricks model serving, we will now be setting env vars with `DB` in place of `DATABRICKS`, and MlFlow has to check for the updated env vars.

<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/prithvikannan/mlflow/pull/12260?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12260/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12260
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Update the `DatabricksModelServingConfigProvider._get_databricks_model_serving_config` logic to read from new `DB_MODEL_SERVING_HOST_URL` flag if available, otherwise fallback to the original `DATABRICKS_MODEL_SERVING_HOST_URL`

Additionally, we are using `DATABRICKS_DEPENDENCY_OAUTH_CACHE` and `DATABRICKS_DEPENDENCY_OAUTH_CACHE_EXIRY_TS ENV` vars to maintain a cache for the the credential. 

These are written inside MLflow `_get_databricks_model_serving_config` after reading the oauth token from file, and read from the same function. Since we control both the reading and writing paths here, we can safely rename these to `DB_DEPENDENCY_...`

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/legacy_databricks_cli/configure/provider.py [hunk 1]
```diff
@@ -325,6 +325,12 @@ def _get_databricks_model_serving_config():
         OAUTH_CACHE_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE"
         OAUTH_CACHE_EXPIRATION_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE_EXIRY_TS"
         MODEL_SERVING_HOST_ENV_VAR = "DATABRICKS_MODEL_SERVING_HOST_URL"
+        DB_MODEL_SERVING_HOST_ENV_VAR = "DB_MODEL_SERVING_HOST_URL"
+
+        # read from DB_MODEL_SERVING_HOST_ENV_VAR if available otherwise MODEL_SERVING_HOST_ENV_VAR
+        host = os.environ.get(
+            DB_MODEL_SERVING_HOST_ENV_VAR, os.environ.get(MODEL_SERVING_HOST_ENV_VAR)
+        )
```
Comment (BenWilson2) [inline mlflow/legacy_databricks_cli/configure/provider.py:333]:
```suggestion
        host = os.environ.get(DB_MODEL_SERVING_HOST_ENV_VAR) or os.environ.get(MODEL_SERVING_HOST_ENV_VAR)
```
minor nit for extensibility and readability

Comment (harupy) [inline mlflow/legacy_databricks_cli/configure/provider.py:333]: can you update this test to ensure this works fine for both `DB_MODEL_SERVING_HOST_ENV_VAR` and `MODEL_SERVING_HOST_ENV_VAR`?

https://github.com/mlflow/mlflow/blob/86ca89a804136413a918bd4a377b478b9d443fe9/tests/utils/test_databricks_utils.py#L88-L90

**Review by harupy — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/12260/files#r1628576647 is addressed

File: tests/utils/test_databricks_utils.py [hunk 2]
```diff
@@ -127,11 +127,27 @@ def test_get_model_dependency_oauth_token_model_serving_throws():
         databricks_utils.get_model_dependency_oauth_token()
 
 
-def test_databricks_params_model_serving_oauth_cache(monkeypatch, oauth_file):
+def test_databricks_params_model_serving_oauth_cache_databricks_env_var(monkeypatch, oauth_file):
```
Comment (harupy) [inline tests/utils/test_databricks_utils.py:130]:
```suggestion
@pytest.mark.parametrize("env_var", [...])
def test_databricks_params_model_serving_oauth_cache_databricks_env_var(monkeypatch, oauth_file, env_var):
```

can we `parametrize` the env var name?

**Review by B-Step62 — APPROVED** [review]
LGTM!

(If I can ask a bit of favor, it would be helpful if you can leave quick 2-3 sentences in PR description about what is the issue of `DATABRICKS_` env name i.e. the JSL thing. It's more for future reference, someone may hit the same issue and refer previous commits for the context).

File: mlflow/legacy_databricks_cli/configure/provider.py [hunk 3]
```diff
@@ -322,9 +322,15 @@ def _get_databricks_model_serving_config():
         # at any point in time but refresh at higher rate of every 5 min here to be safe
         # and conform with refresh logic for Brickstore tables.
         OAUTH_CACHE_REFRESH_DURATION_SEC = 5 * 60
-        OAUTH_CACHE_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE"
-        OAUTH_CACHE_EXPIRATION_ENV_VAR = "DATABRICKS_DEPENDENCY_OAUTH_CACHE_EXIRY_TS"
+        OAUTH_CACHE_ENV_VAR = "DB_DEPENDENCY_OAUTH_CACHE"
+        OAUTH_CACHE_EXPIRATION_ENV_VAR = "DB_DEPENDENCY_OAUTH_CACHE_EXIRY_TS"
```
Comment (harupy) [inline mlflow/legacy_databricks_cli/configure/provider.py:326]: Is `EXIRY` a typo?

Comment (prithvikannan) [reply mlflow/legacy_databricks_cli/configure/provider.py:326]: haha yea -- let me fix

**Comment by prithvikannan** [comment]
> LGTM!
> 
> (If I can ask a bit of favor, it would be helpful if you can leave quick 2-3 sentences in PR description about what is the issue of `DATABRICKS_` env name i.e. the JSL thing. It's more for future reference, someone may hit the same issue and refer previous commits for the context).

yea totally. @bbqiu will add

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Comment by bbqiu** [comment]
updated the PR description! thank you for doing this prithvi

**Pull request merged**